flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliTableauResultView.java

```diff
@@ -122,14 +137,83 @@ private void checkAndCleanUpQuery(boolean cleanUpQuery) {
     private void printBatchResults(AtomicInteger receivedRowCount) {
         final List<RowData> resultRows = waitBatchResults();
         receivedRowCount.addAndGet(resultRows.size());
+
+        if (printBatchEmptySet(resultRows)) return;
+        printBatchTable(resultRows);
+        printBatchFooter(resultRows.size());
+    }
+
+    /**
+     * Print "Empty set" if the given <code>resultRows</code> is empty, otherwise will do nothing.
+     *
+     * @return false if <code>resultRows</code> is not empty
+     */
+    public boolean printBatchEmptySet(List<RowData> resultRows) {
```

Comment:
nit: pubilc -> private.

The method name describes it is an action to do something. At the first glance, I think it should retun void. How about

```
 if (resultRows.isEmpty()) {
        printBatchEmptySet();
} else {
        printBatchTable(resultRows);
        printBatchFooter(resultRows.size());
}
```